### PR TITLE
fix: adding csi link and correct some words used to clarify access mode.

### LINKED ISSUE #18714: Issue with k8s.io/docs/concepts/storage/persistent-volumes/
**This is a Bug Report**

**Problem:**

Are below statements correct? There are some discussions in the Internet that the "node" should be the "pod", i.e. ReadWriteOnce only allows a volume be mounted as read-write by a single pod. 

ReadWriteOnce – the volume can be mounted as read-write by a single node
ReadOnlyMany – the volume can be mounted read-only by many nodes
ReadWriteMany – the volume can be mounted as read-write by many nodes

https://github.com/kubernetes/kubernetes/issues/60903

**Proposed Solution:**

**Page to Update:**
https://kubernetes.io/...

### PR DESCRIPTION
**Pull request by Colstuwjx**
This PR is going to fix #18714 , the `PV \ PVC \ Pod Volume Mount` such terms is not clear enough for our users, cause some confusions about the accessMode.

Within this PR, I'd like to change some words used to clarify the process of volume mount, and hope it could make our users really understand how pv/pvc designed and how to use it.

Thanks.

### CODE REVIEW

**Comment by Colstuwjx** [comment]
As @msau42 [said](https://github.com/kubernetes/kubernetes/issues/60903#issuecomment-377086770), accessMode describes `node attach` behavior rather than `pod mount`, but it would change lots of content inside this page. So I choose to change some words used to clarify the process of volume mount.

It may needs both tech and content review, but I think PR for changing the words between each stage about PV \ PVC \ Volume mount \ Node attach is really required, since it made our users confused for many times.

Thanks.

File: content/en/docs/concepts/storage/persistent-volumes.md [hunk 1]
```diff
@@ -75,9 +75,9 @@ A user creates, or in the case of dynamic provisioning, has already created, a `
 
 Claims will remain unbound indefinitely if a matching volume does not exist. Claims will be bound as matching volumes become available. For example, a cluster provisioned with many 50Gi PVs would not match a PVC requesting 100Gi. The PVC can be bound when a 100Gi PV is added to the cluster.
 
-### Using
+### Mounting and Using
```
Comment (sftim) [inline content/en/docs/concepts/storage/persistent-volumes.md]: nit:
```suggestion
### Mounting and Using {#using}
```

will preserve links to this page fragment.

File: content/en/docs/concepts/storage/persistent-volumes.md [hunk 2]
```diff
@@ -75,9 +75,9 @@ A user creates, or in the case of dynamic provisioning, has already created, a `
...
 
-Pods use claims as volumes. The cluster inspects the claim to find the bound volume and mounts that volume for a Pod. For volumes that support multiple access modes, the user specifies which mode is desired when using their claim as a volume in a Pod.
+Pods use claims as volumes. The cluster inspects the claim to find the bound volume and mounts that volume for a Pod. Once the pvc which bound with pv has been used by pods, it will do actually volume mount on node. For volumes that support multiple access modes, the user specifies which mode is desired when using their claim as a volume in a Pod.
```
Comment (sftim) [inline content/en/docs/concepts/storage/persistent-volumes.md]: @Colstuwjx this new wording doesn't seem to be idiomatic English. Can you revise it?

File: content/en/docs/concepts/storage/persistent-volumes.md [hunk 3]
```diff
@@ -339,7 +338,7 @@ In the CLI, the access modes are abbreviated to:
 * ROX - ReadOnlyMany
 * RWX - ReadWriteMany
 
-> __Important!__ A volume can only be mounted using one access mode at a time, even if it supports many.  For example, a GCEPersistentDisk can be mounted as ReadWriteOnce by a single node or ReadOnlyMany by many nodes, but not at the same time.
+> __Important!__ A volume can only be mounted using one access mode at a time, even if it supports many.  For example, a GCEPersistentDisk can be mounted as ReadWriteOnce by a single node or ReadOnlyMany by many nodes, but not at the same time. Besides, it will be limited once a volume has been mounted on a single node with `ReadWriteOnce`, and pod which using this volume will be [checked](https://kubernetes.io/docs/concepts/scheduling/kube-scheduler/#filtering) whether it fit the volume mount in scheduling stage.
```
Comment (sftim) [inline content/en/docs/concepts/storage/persistent-volumes.md]: If you're rewriting this, consider using a `{{< caution >}}` shortcode.

File: content/en/docs/concepts/storage/persistent-volumes.md (hunk 2 above)
Comment (Colstuwjx) [reply content/en/docs/concepts/storage/persistent-volumes.md]: Em.. I'm not native speaker, could you please give me some suggest changes?

File: content/en/docs/concepts/storage/persistent-volumes.md (hunk 3 above)
Comment (Colstuwjx) [reply content/en/docs/concepts/storage/persistent-volumes.md]: Okay, I'll have a try.

**Comment by Colstuwjx** [comment]
/hold
since the PR requires technical review.

File: content/en/docs/concepts/storage/persistent-volumes.md (hunk 2 above)
Comment (sftim) [reply content/en/docs/concepts/storage/persistent-volumes.md]: How about:
```suggestion
Pods use claims as volumes. The cluster inspects the claim to find a matching PersistentVolume for the Pod, and to check that the Pod is entitled to use it. If the claim is valid and the storage is available, the kubelet sets up storage access so that the PersistentVolume is available to containers in the Pod (either as mounted storage or as a block device). The PersistentVolumeClaim also defines whether the kubelet should mount the volume read-only or read-write.
```
```

**Comment by sftim** [comment]
BTW I'm not sure this needs a /hold
Approvers will wait for technical review if a PR merits it.

**Comment by Colstuwjx** [comment]
@sftim I have been rebased and revised some content according the comments. However, I still think the pv lifecycle terminologies: Provisioning -> Binding -> Mounting and Using are not good to understand.

See [CSI Lifecycle](https://github.com/container-storage-interface/spec/blob/master/spec.md#volume-lifecycle), I think it should be sth like Provisioning -> Publishing ( or Attaching/Binding ) -> Staging (or Mounting) -> Using, we should unify these terms and make these terms could map to CSI terms.

If you agree with this, I'd like to propose another PR to unify these terms, and this PR just revise some words to clarify the access mode, as well as the title said.

It's also the reason that I think this PR need tech reviews, we need to make our users understand the pv binding and mounting process, and I'm not sure whether my understand is correct, so I'd like to get confirm from tech reviewers.

Thanks.

File: content/en/docs/concepts/storage/persistent-volumes.md [hunk 4]
```diff
@@ -353,7 +352,11 @@ In the CLI, the access modes are abbreviated to:
 * ROX - ReadOnlyMany
 * RWX - ReadWriteMany
 
-> __Important!__ A volume can only be mounted using one access mode at a time, even if it supports many.  For example, a GCEPersistentDisk can be mounted as ReadWriteOnce by a single node or ReadOnlyMany by many nodes, but not at the same time.
+{{< caution >}}
+
+A volume can only be mounted using one access mode at a time, even if it supports many.  For example, a GCEPersistentDisk can be mounted as ReadWriteOnce by a single node or ReadOnlyMany by many nodes, but not at the same time. Besides, it will be limited once a volume has been mounted on a single node with `ReadWriteOnce`, and pod which using this volume will be [checked](https://kubernetes.io/docs/concepts/scheduling/kube-scheduler/#filtering) whether it fit the volume mount in scheduling stage.
```
Comment (sftim) [inline content/en/docs/concepts/storage/persistent-volumes.md:357]: nit: “whether it fit the volume mount in scheduling stage” doesn't seem like idiomatic English. I'm also not sure what meaning you want to convey here @Colstuwjx.

Comment (Colstuwjx) [reply content/en/docs/concepts/storage/persistent-volumes.md]: Okay, I removed this line.

File: content/en/docs/concepts/storage/persistent-volumes.md [hunk 5]
```diff
@@ -353,7 +352,11 @@ In the CLI, the access modes are abbreviated to:
 * ROX - ReadOnlyMany
 * RWX - ReadWriteMany
 
-> __Important!__ A volume can only be mounted using one access mode at a time, even if it supports many.  For example, a GCEPersistentDisk can be mounted as ReadWriteOnce by a single node or ReadOnlyMany by many nodes, but not at the same time.
+{{< caution >}}
+
+A volume can only be mounted using one access mode at a time, even if it supports many.  For example, a GCEPersistentDisk can be mounted as ReadWriteOnce by a single node or ReadOnlyMany by many nodes, but not at the same time. Besides, it will be limited once a volume has been mounted on a single node with `ReadWriteOnce`.
```
Comment (tengqm) [inline content/en/docs/concepts/storage/persistent-volumes.md:357]: > Besides, it will be limited once a volume has been mounted on a single node with `ReadWriteOnce`.

This sentence is hard to parse.

**Comment by sftim** [comment]
There's room for more polish, but let's get this past technical review and see if it's good enough to merge.

@Colstuwjx if you'd rebase this against _master_ we can ask SIG Storage for a review.
(I'm asking for a rebase because the preview looks very different to the live page - this CI system doesn't do a test merge and then build).

File: content/en/docs/concepts/storage/persistent-volumes.md (hunk 5 above)
Comment (Colstuwjx) [reply content/en/docs/concepts/storage/persistent-volumes.md:357]: Do you have suggestion about this?

**Comment by Colstuwjx** [comment]
@sftim I've been rebased! The reason why I propose this PR is that our website is not well documented for the pv and csi lifecycle.

As [my comment](https://github.com/kubernetes/website/issues/18714#issuecomment-575084522) said, I prefer some words like `Provisioning`, `Binding`, `Attaching`, `Using`, rather than `Mount` such ambiguous words.

**Comment by sftim** [comment]
/hold cancel

@kubernetes/sig-storage-pr-reviews does this change leave https://k8s.io/docs/concepts/storage/persistent-volumes/ better than it finds it? (see [preview](https://deploy-preview-18745--kubernetes-io-master-staging.netlify.app/docs/concepts/storage/persistent-volumes/))

**Comment by sftim** [comment]
Hi @kubernetes/sig-storage-pr-reviews 
Have you got time to take a look at this PR?

**Comment by jimangel** [comment]
:wave: @kubernetes/sig-storage-pr-reviews

**Comment by Colstuwjx** [comment]
It seems that less people care about this PR and the proposed changes, so I'd like to close this PR, if someone think it's reasonable change, feel free to reopen this or create another better one PR.

File: content/en/docs/concepts/storage/persistent-volumes.md (hunk 5 above)
Comment (xing-yang) [reply content/en/docs/concepts/storage/persistent-volumes.md:357]: I would suggest drop the last sentence "Besides, it will be limited once a volume has been mounted on a single node with `ReadWriteOnce`."

File: content/en/docs/concepts/storage/persistent-volumes.md [hunk 6]
```diff
@@ -71,9 +71,9 @@ A user creates, or in the case of dynamic provisioning, has already created, a P
 
 Claims will remain unbound indefinitely if a matching volume does not exist. Claims will be bound as matching volumes become available. For example, a cluster provisioned with many 50Gi PVs would not match a PVC requesting 100Gi. The PVC can be bound when a 100Gi PV is added to the cluster.
 
-### Using
+### Mounting and Using {#using}
 
-Pods use claims as volumes. The cluster inspects the claim to find the bound volume and mounts that volume for a Pod. For volumes that support multiple access modes, the user specifies which mode is desired when using their claim as a volume in a Pod.
+Pods use claims as volumes. The cluster inspects the claim to find a matching PersistentVolume for the Pod, and to check that the Pod is entitled to use it. If the claim is valid and the storage is available, the kubelet sets up storage access so that the PersistentVolume is available to containers in the Pod (either as mounted storage or as a block device). The PersistentVolumeClaim also defines whether the kubelet should mount the volume read-only or read-write.
```
Comment (xing-yang) [inline content/en/docs/concepts/storage/persistent-volumes.md:76]: I think this part "find the bound volume and mounts that volume for a Pod" should still be mentioned.

Comment (xing-yang) [reply content/en/docs/concepts/storage/persistent-volumes.md:76]: CC @msau42

Comment (kbhawkey) [reply content/en/docs/concepts/storage/persistent-volumes.md:76]: Hello @Colstuwjx .
Are you able to add back the original sentence or is there a question about `the Pod is entitled to use it`?
Ping @msau42 , Would you have time to review the changes? Thank you.

File: content/en/docs/concepts/storage/persistent-volumes.md (hunk 5 above)
Comment (kbhawkey) [reply content/en/docs/concepts/storage/persistent-volumes.md:357]: @Colstuwjx , The `caution` shortcode is a great addition. Would you remove the last sentence?
I am not sure if it is clear what is `limited`.

**Comment by zacharysarah** [comment]
@kubernetes/sig-storage-pr-reviews It's been 2+ months with no response. Please review!

**Comment by zacharysarah** [comment]
@Colstuwjx 👋  Please feel free to `/reopen` when you're ready to respond to [reviewer feedback](https://github.com/kubernetes/website/pull/18745#discussion_r460869335). 

/close
